packages/web/components/control/language-select.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback } from "react";
+import { FunctionComponent } from "react";
+import { useBooleanWithWindowEvent, useWindowSize } from "../../hooks";
+import { observer } from "mobx-react-lite";
+import { useStore } from "../../stores";
+import Image from "next/image";
+import { MenuDropdownIcon } from "./menu-dropdown-icon/menu-dropdown-icon";
+import { MenuDropdownIconItem } from "./types";
+import { MenuOptionsIconModal } from "../../modals";
+import { useTranslation } from "react-multi-lang";
+
+export type LanguageSelectProps = {
+  options: MenuDropdownIconItem[];
+};
+
+export const LanguageSelect: FunctionComponent<LanguageSelectProps> = observer(
+  ({ options }) => {
+    const { isMobile } = useWindowSize();
+    const { userSettings } = useStore();
+    const t = useTranslation();
+    const [dropdownOpen, setDropdownOpen] = useBooleanWithWindowEvent(false);
+    const currentLanguage =
+      userSettings.getUserSettingById("language")?.state.language;
+    const currentOption = options.find(
+      (option) => option.value === currentLanguage
+    );
+
+    const onSelect = useCallback(
+      ({ value }: { value: string }) => {
+        setDropdownOpen(false);
+
+        userSettings
+          .getUserSettingById("language")
+          ?.setState({ language: value });
+      },
+      [setDropdownOpen]
+    );
+
+    return (
+      <div className="relative">
+        <button
+          className="flex items-center border border-osmoverse-200 rounded-xl bg-osmoverse-900"
```

Comment:
I think the default/rest state has a different border. This osmoverse-200 border seems to be for when the dropdown is open.

See rest state design:
https://www.figma.com/file/kYSFzlWoiMFrqkF46dtZQV/Osmosis-Design-Guidelines-2022?node-id=528%3A3785